desktop/components/react/utils/template_renderer.js

```diff
@@ -0,0 +1,41 @@
+import invariant from 'invariant'
+import jade from 'jade'
+import path from 'path'
+
+import {
+  isArray,
+  isString
+} from 'underscore'
+
+export default function templateRenderer (templates, options = {}) {
+  const isValid = isArray(templates) || isString(templates)
+
+  invariant(isValid,
+    '(lib/template_renderer.js) ' +
+    'Error rendering templates: `templates` must be a string or array of ' +
+    'strings representing the path to the template'
+  )
+
+  templates = isArray(templates)
```

Comment:
Ah yes, will fix :) that will def make things a little simpler and more symmetric. And yeah, that's interesting re `Id` functor! I've seen that around and kinda understood it, but have never known the particulars. Tyty.